bindgen-tests/tests/expectations/tests/libclang-5/constant-evaluate.rs

```diff
@@ -21,5 +21,13 @@ pub const BAZ: ::std::os::raw::c_longlong = 24;
 pub const fuzz: f64 = 51.0;
 pub const BAZZ: ::std::os::raw::c_char = 53;
 pub const WAT: ::std::os::raw::c_char = 0;
-pub const bytestring: &[u8; 4usize] = b"Foo\0";
-pub const NOT_UTF8: [u8; 5usize] = [240u8, 40u8, 140u8, 40u8, 0u8];
+#[allow(unsafe_code)]
```

Comment:
Some people care more about having a clean [`cargo geiger`](https://github.com/rust-secure-code/cargo-geiger) output than others. Is `#[allow(unsafe_code)]` going to make some users unhappy?

I understand that common use cases for `bindgen` probably imply buried `unsafe` code a good proportion of the time, so maybe this is a non-issue.

This is a policy question for people like @emilio, @pvdrz, @amanjeev.

Reply:
Does `cargo geiger` also treat `unsafe` code in `std` as unsafe, or is there some barrier at which point it considers code safe? E.g., would `CStr::from_bytes_with_nul(BYTES).unwrap()` be considered safe, even though it uses `CStr::from_bytes_with_nul_unchecked` internally?

For context, this is as safe as it gets: 

We only generate a `CStr` once we constructed one. Even if we didn't, `CStr::from_bytes_with_nul_unchecked` does panic and is not actually `_unchecked` in `const` contexts.

Reply:
I'd say that it is better to keep this behind a flag/method named `--use-cstr` (or something better :P).

This flag should be disabled by default. In that way we can smoothly introduce this feature and make it the default eventually.

@emilio  and @amanjeev might differ